classes/services/PKPSiteService.inc.php

```diff
@@ -0,0 +1,333 @@
+					}
+				}
+			}
+		});
+
+		// Ensure the theme plugin is installed and enabled
+		$validator->after(function($validator) use ($props) {
+			if (!empty($props['themePluginPath']) && !$validator->errors()->get('themePluginPath')) {
+				$plugins = \PluginRegistry::loadCategory('themes', true);
+				$found = false;
+				foreach ($plugins as $plugin) {
+					if ($props['themePluginPath'] === $plugin->getDirName()) {
+						$found = true;
+						break;
+					}
+				}
+				if (!$found) {
+					$validator->errors()->add('themePluginPath', __('manager.setup.theme.notFound'));
+				}
+			}
+		});
+
+		if ($validator->fails()) {
+			$errors = $schemaService->formatValidationErrors($validator->errors(), $schemaService->get(SCHEMA_SITE), $allowedLocales);
+		}
+
+		\HookRegistry::call('Site::validate', array(&$errors, $props, $allowedLocales, $primaryLocale));
+
+		return $errors;
+	}
+
+	/**
+	 * Edit the site
+	 *
+	 * This does not check if the user is authorized to edit the site, or validate or sanitize
+	 * the new content.
+	 *
+	 * @param $site Context The context to edit
+	 * @param $params Array Key/value array of new data
+	 * @param $request Request
+	 * @return Site
+	 */
+	public function editSite($site, $params, $request) {
+		$siteDao = DAORegistry::getDAO('SiteDAO');
+
+		// Move uploaded files into place and update the params
+		$userId = $request->getUser() ? $request->getUser()->getId() : null;
+		$supportedLocales = $site->getSupportedLocales();
+		if (array_key_exists('pageHeaderTitleImage', $params)) {
+			foreach ($supportedLocales as $localeKey) {
+				if (!array_key_exists($localeKey, $params['pageHeaderTitleImage'])) {
+					continue;
+				}
+				$params['pageHeaderTitleImage'][$localeKey] = $this->_saveFileParam($site, $params['pageHeaderTitleImage'][$localeKey], 'pageHeaderTitleImage', $userId, $localeKey, true);
+			}
+		}
+		if (array_key_exists('styleSheet', $params)) {
+			$params['styleSheet'] = $this->_saveFileParam($site, $params['styleSheet'], 'styleSheet', $userId);
+		}
+
+		$newSite = $siteDao->newDataObject();
+		$newSite->_data = array_merge($site->_data, $params);
+
+		\HookRegistry::call('Site::edit', array($newSite, $site, $params, $request));
+
+		$siteDao->updateObject($newSite);
+		$newSite = $this->getSite();
+
+		return $newSite;
+	}
+
+	/**
+	 * Move a temporary file to the site's public directory
+	 *
+	 * @param $context Context
+	 * @param $temporaryFile TemporaryFile
+	 * @param $fileNameBase string Unique identifier to use for the filename. The
+	 *  Extension and locale will be appended.
+	 * @param $userId int ID of the user who uploaded the temporary file
+	 * @param $localeKey string Example: en_US. Leave empty for a file that is
+	 *  not localized.
+	 * @return string|boolean The new filename or false on failure
+	 */
+	public function moveTemporaryFile($context, $temporaryFile, $fileNameBase, $userId, $localeKey = '') {
```

Comment:
I see this duplicated too... Maybe an opportunity for a helper in `TemporaryFileManager`?

Reply:
I was thinking that I would wait and see what things looked like with submission, review and discussion files before figuring out what/how to extract. Site/Context are pretty similar, but the others will be different, and I think a better base method or helper methods to service the classes might come into view? 

Happy to start that now, though, if you want. I was hesitant to abstract too much because Site is a really unique object in the system (no context, only ever one, etc).